### PR TITLE
service_discovery: add telemetry

### PR DESCRIPTION
**Pull request by rarguelloF**
### What does this PR do?

Adds telemetry for the service_discovery check.

### Motivation

Get data on how the service_discovery check is performing/behaving.

### CODE REVIEW

File: pkg/collector/corechecks/servicediscovery/servicediscovery.go [hunk 1]
```diff
@@ -130,7 +153,121 @@ func (c *Check) Configure(senderManager sender.SenderManager, _ uint64, instance
 
 // Run executes the check.
 func (c *Check) Run() error {
-	return c.os.DiscoverServices()
+	start := time.Now()
+	defer func() {
+		end := time.Now()
+		diff := end.Sub(start).Seconds()
```
Comment (jonbodner) [inline pkg/collector/corechecks/servicediscovery/servicediscovery.go:159]: don't need to define `end`, just use `time.Since(start).Seconds()`

**Review by jonbodner — CHANGES_REQUESTED** [review]

**Comment by jonbodner** [comment]
Additional code coverage is needed, too.

**Review by jonbodner — APPROVED** [review]
This should have a second approver (@amarziali has been asked) since I made modifications to the PR. But I approve of my modifications 😄

File: pkg/collector/corechecks/servicediscovery/servicediscovery.go [hunk 2]
```diff
@@ -128,7 +153,124 @@ func (c *Check) Configure(senderManager sender.SenderManager, _ uint64, instance
...
+	}
+	for _, svcs := range runningServicesByName {
+		if len(svcs) <= 1 {
+			continue
+		}
+		for _, svc := range svcs {
+			if c.sentRepeatedEventPIDs[svc.process.PID] {
+				continue
+			}
+			err := fmt.Errorf("found repeated service name: %s", svc.meta.Name)
```
Comment (amarziali) [inline pkg/collector/corechecks/servicediscovery/servicediscovery.go:189]: That situation can happen quite often

File: pkg/collector/corechecks/servicediscovery/servicediscovery.go [hunk 3]
```diff
@@ -128,7 +153,124 @@ func (c *Check) Configure(senderManager sender.SenderManager, _ uint64, instance
...
+		}
+	}
+
+	potentialNames := map[string]bool{}
+	for _, p := range disc.potentials {
+		potentialNames[p.meta.Name] = true
+	}
+
+	// group events by name in order to find repeated events for the same service name.
+	eventsByName := make(eventsByNameMap)
```
Comment (amarziali) [inline pkg/collector/corechecks/servicediscovery/servicediscovery.go:206]: we group by name but that key is not guarantee to be unique. Is the logic correct in this way or should we have a better key?

**Review by amarziali — APPROVED** [review]
It looks ok. I left a couple of questions about the logic

File: pkg/collector/corechecks/servicediscovery/servicediscovery.go (hunk 2 above)
Comment (jonbodner) [reply pkg/collector/corechecks/servicediscovery/servicediscovery.go:189]: What scenarios make it likely? We do want to know how many duplicate service names we generate so we can see if we need to tweak the service name generation code.

File: pkg/collector/corechecks/servicediscovery/servicediscovery.go (hunk 3 above)
Comment (jonbodner) [reply pkg/collector/corechecks/servicediscovery/servicediscovery.go:206]: eventsByNameMap has a key of name and a value that's a struct containing slices, each one representing a service state. So if there's a collision, we should see all the processes with the same name in all states.

**Pull request merged**